libyul/backends/evm/SSAControlFlowGraph.h

```diff
@@ -0,0 +1,229 @@
+
+		};
+		struct FunctionReturn
+		{
+			langutil::DebugData::ConstPtr debugData;
+			std::vector<ValueId> returnValues;
+		};
+		struct Terminated {};
+		langutil::DebugData::ConstPtr debugData;
+		std::set<BlockId> entries;
+		std::set<ValueId> phis;
+		std::vector<Operation> operations;
+		std::variant<MainExit, Jump, ConditionalJump, JumpTable, FunctionReturn, Terminated> exit = MainExit{};
+		template<typename Callable>
+		void forEachExit(Callable&& _callable)
+		{
+			if (auto* jump = std::get_if<Jump>(&exit))
+				_callable(jump->target);
+			else if (auto* conditionalJump = std::get_if<ConditionalJump>(&exit))
+			{
+				_callable(conditionalJump->nonZero);
+				_callable(conditionalJump->zero);
+			}
+			else if (auto* jumpTable = std::get_if<JumpTable>(&exit))
+			{
+				for (auto _case: jumpTable->cases | ranges::views::values)
+					_callable(_case);
+				_callable(jumpTable->defaultCase);
+			}
+		}
+	};
+	BlockId makeBlock(langutil::DebugData::ConstPtr _debugData)
+	{
+		BlockId blockId { m_blocks.size() };
+		m_blocks.emplace_back(BasicBlock{std::move(_debugData), {}, {}, {}, BasicBlock::Terminated{}});
+		return blockId;
+	}
+	BasicBlock& block(BlockId _id) { return m_blocks.at(_id.value); }
+	BasicBlock const& block(BlockId _id) const { return m_blocks.at(_id.value); }
+	size_t numBlocks() const { return m_blocks.size(); }
+private:
+	std::vector<BasicBlock> m_blocks;
+public:
+	struct LiteralValue {
+		langutil::DebugData::ConstPtr debugData;
+		u256 value;
+	};
+	struct VariableValue {
+		langutil::DebugData::ConstPtr debugData;
+		BlockId definingBlock;
+	};
+	struct PhiValue {
+		langutil::DebugData::ConstPtr debugData;
+		BlockId block;
+		std::vector<ValueId> arguments;
+	};
+	struct UnreachableValue {};
+	using ValueInfo = std::variant<UnreachableValue, VariableValue, LiteralValue, PhiValue>;
+	ValueInfo& valueInfo(ValueId _var)
+	{
+		yulAssert(_var.value < m_valueInfos.size());
+		return m_valueInfos.at(_var.value);
+	}
+	ValueInfo const& valueInfo(ValueId _var) const
+	{
+		yulAssert(_var.value < m_valueInfos.size());
+		return m_valueInfos.at(_var.value);
+	}
+	ValueId newPhi(BlockId _definingBlock)
+	{
+		ValueId id { m_valueInfos.size() };
+		auto block = m_blocks.at(_definingBlock.value);
+		m_valueInfos.emplace_back(PhiValue{debugDataOf(block), _definingBlock, {}});
+		return id;
+	}
+	ValueId newVariable(BlockId _definingBlock)
+	{
+		ValueId id { m_valueInfos.size() };
+		auto block = m_blocks.at(_definingBlock.value);
+		m_valueInfos.emplace_back(VariableValue{debugDataOf(block), _definingBlock});
+		return id;
+	}
+	ValueId unreachableValue()
+	{
+		if (!m_unreachableValue)
+		{
+			m_unreachableValue = ValueId { m_valueInfos.size() };
+			m_valueInfos.emplace_back(UnreachableValue{});
+		}
+		return *m_unreachableValue;
+	}
+	ValueId newLiteral(langutil::DebugData::ConstPtr _debugData, u256 _value)
+	{
+		auto [it, inserted] = m_literals.emplace(std::make_pair(_value, ValueId{m_valueInfos.size()}));
```

Comment:
```suggestion
		auto [it, inserted] = m_literals.emplace(_value, ValueId{m_valueInfos.size()});
```
I think `emplace` should be able to perfectly forward the arguments.

Reply:
Yes, we changed that in another branch haha